src/tls.rs

```diff
@@ -1,9 +1,22 @@
+use std::{borrow::BorrowMut, sync::LazyLock};
+
+use dashmap::{try_result::TryResult, DashMap};
+use rustls::ServerConfig;
+
+static SERVER_CONFIG_CACHE: LazyLock<DashMap<String, rustls::ServerConfig>> = LazyLock::new(|| DashMap::new());
```

Comment:
Caches should be held for each `MitmProxy` instance to support multiple proxy servers with different root certs.

Currently, it may cache a cert for different root cert